test/studies/prk/DGEMM2/dgemm.chpl

```diff
@@ -0,0 +1,115 @@
+use Time;
+use BlockDist;
+use LinearAlgebra;
+
+config type dtype = real;
+
+config param useBlockDist = false;
+
+config const order = 10,
+             epsilon = 1e-8,
+             iterations = 100,
+             windowSize = 10,
+             debug = false,
+             validate = true,
+             correctness = false; // being run in start_test
+
+const vecRange = 0..#order;
+const matrixSpace = {vecRange, vecRange};
+
+const matrixDom = matrixSpace dmapped if useBlockDist then
+                      new dmap(new Block(boundingBox=matrixSpace)) else
+                      defaultDist;
+
+var A = Matrix(matrixDom, dtype),
+    B = Matrix(matrixDom, dtype),
+    C = Matrix(matrixDom, dtype);
+
+forall (i,j) in matrixDom {
+  A[i,j] = j;
+  B[i,j] = j;
+  C[i,j] = 0;
+}
+
+const nTasksPerLocale = here.maxTaskPar;
+
+if !correctness {
+  writeln("Chapel Dense matrix-matrix multiplication");
+  writeln("Max parallelism      =   ", nTasksPerLocale);
+  writeln("Matrix order         =   ", order);
+  writeln("Window size      =   ", if windowSize>0 then 
+      windowSize:string
+      else "N/A");
+  writeln("Number of iterations =   ", iterations);
+  writeln();
+}
+
+const refChecksum = (iterations+1) *
+    (0.25*order*order*order*(order-1.0)*(order-1.0));
+
+var t = new Timer();
+
+private inline iter block(indexes, window : int) {
+  var lowVal = indexes.low;
+  while (true) {
+    var highVal = lowVal + indexes.stride * window - 1;
```

Comment:
for the benchmarking purposes, I think we can drop `stride` logic here. It begs the question whether the rest of this logic is well adjusted for strides, alignments etc.